test/jdk/java/lang/ProcessBuilder/ProcessStartLoggingTest.java

```diff
@@ -0,0 +1,188 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import java.util.logging.StreamHandler;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.ParameterizedTest;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/*
+ * @test
+ * @summary verify logging of ProcessBuilder.start()
+ * @run junit/othervm ProcessStartLoggingTest
+ */
+public class ProcessStartLoggingTest {
+
+    private static final String TEST_JDK = System.getProperty("test.jdk");
+    private static final String TEST_SRC = System.getProperty("test.src");
+
+    private static Object HOLD_LOGGER;
+
+    /**
+     * Launch a process with the arguments.
+     * @param args 1 or strings passed directly to ProcessBuilder as command and arguments.
+     */
+    public static void main(String[] args) throws InterruptedException {
+        if (System.getProperty("ThrowingHandler") != null) {
+            HOLD_LOGGER = ProcessStartLoggingTest.ThrowingHandler.installHandler();
+        }
+        String directory = System.getProperty("directory");
+        try {
+            ProcessBuilder pb = new ProcessBuilder(args);
+            pb.directory((directory == null) ? null : new File(directory));
+            Process p = pb.start();
+            int status = p.waitFor();
+            if (status != 0) {
+                System.out.println("exitValue: " + status);
+            }
+        } catch (IOException ioe) {
+            System.out.println("ProcessBuilder.start() threw IOException: " + ioe);
+        }
+    }
+
+    /**
+     * Test various log level settings, and none.
+     * @return a stream of arguments for parameterized test
+     */
+    private static Stream<Arguments> logParamProvider() {
+        File nullDirectory = null;
+        File thisDirectory = new File(".");
+
```

Comment:
I started to heavily use different `@test` sections with speaking names instead of running a bunch of test in sequence; the advantage is better parallelization of tests, that I can omit printing out the test name manually (if the test name itself is speaking), and the ability to start individual tests manually. It does come with lots of test sections though.

Reply:
Speakable names can be a plus but I'll stick to the parameterized version for now.
Are you refering to JUnit sections or Jtreg sections?

Reply:
jtreg sections, as in https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/gtest/MetaspaceGtests.java. I find them very practical, I just wish there were a way to condense them.